tests/Microsoft.Identity.Test.Integration.Win8/DeviceAuthenticationTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Threading;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.net45.HeadlessTests
+{
+    [TestClass]
+    public class DeviceAuthenticationTests
+    {
+        private const string _claims = "{\"access_token\":{\"deviceid\":{\"essential\":true}}}";
+        private const string _publicClientId = "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0";
+        private const string _deviceAuthuser = "idlabca@msidlab8.onmicrosoft.com";
+
+        [TestMethod]
+
+        public void PKeyAuthNonInteractiveTest()
+        {
+            //Arrange
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(_publicClientId)
+                .Build();
+
+            //Act
+            var authResult = msalPublicClient.AcquireTokenByUsernamePassword(new[] { "user.read" }, _deviceAuthuser, new NetworkCredential("", LabUserHelper.FetchUserPassword("msidlab8")).SecurePassword)
+            .WithClaims(JObject.Parse(_claims).ToString())
```

Comment:
will device auth user always have these claims? i'm trying to think of a failure case. what if you try device auth but don't have the claims? or try with the claims but the wrong user?

Reply:
To initiate device auth you add this claim specifically. However, I get your point about the failure case. ill try to add